### ISSUE TITLE
MAKE HTML, bug

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#424]
```
# Sphinx version: 1.0b1
# Docutils version: 0.5 release
# Jinja2 version: 2.4.1
Traceback (most recent call last):
  File "/usr/local/lib/python2.6/dist-packages/Sphinx-1.0b1-py2.6.egg/sphinx/cmdline.py", line 173, in main
    app.build(force_all, filenames)
  File "/usr/local/lib/python2.6/dist-packages/Sphinx-1.0b1-py2.6.egg/sphinx/application.py", line 205, in build
    self.builder.build_update()
  File "/usr/local/lib/python2.6/dist-packages/Sphinx-1.0b1-py2.6.egg/sphinx/builders/__init__.py", line 198, in build_update
    'out of date' &#37; len(to_build))
  File "/usr/local/lib/python2.6/dist-packages/Sphinx-1.0b1-py2.6.egg/sphinx/builders/__init__.py", line 257, in build
    self.finish()
  File "/usr/local/lib/python2.6/dist-packages/Sphinx-1.0b1-py2.6.egg/sphinx/builders/html.py", line 421, in finish
    for pagename, context, template in pagelist:
  File "/usr/local/lib/python2.6/dist-packages/Sphinx-1.0b1-py2.6.egg/sphinx/ext/viewcode.py", line 111, in collect_pages
    lines[end - 1] += '</div>'
IndexError: list index out of range
```

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/424
- Originally reported by: Anonymous
- Originally created at: 2010-05-26T13:14:57.757

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2010-05-26 17:33:26+00:00_

Thanks for the report; should be fixed in <<cset 7e7593c4dd2ba4f34c0787bfb2bc35ce94ced7a7>>.

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2010-05-30 18:07:22+00:00_

Removing version: 1.0b1 (automated comment)

**Comment by shimizukawa** [comment]
_From Kijun Seo on 2010-06-13 01:50:42+00:00_

Actually, I think this still happens even in 1.0b2.

{{{

# Sphinx version: 1.0b2

# Docutils version: 0.6 release

# Jinja2 version: 2.4.1

Traceback (most recent call last):
  File "/usr/local/lib/python2.6/dist-packages/Sphinx-1.0b2-py2.6.egg/sphinx/cmdline.py", line 173, in main
    app.build(force_all, filenames)
  File "/usr/local/lib/python2.6/dist-packages/Sphinx-1.0b2-py2.6.egg/sphinx/application.py", line 205, in build
    self.builder.build_update()
  File "/usr/local/lib/python2.6/dist-packages/Sphinx-1.0b2-py2.6.egg/sphinx/builders/**init**.py", line 198, in build_update
    'out of date' % len(to_build))
  File "/usr/local/lib/python2.6/dist-packages/Sphinx-1.0b2-py2.6.egg/sphinx/builders/**init**.py", line 257, in build
    self.finish()
  File "/usr/local/lib/python2.6/dist-packages/Sphinx-1.0b2-py2.6.egg/sphinx/builders/html.py", line 421, in finish
    for pagename, context, template in pagelist:
  File "/usr/local/lib/python2.6/dist-packages/Sphinx-1.0b2-py2.6.egg/sphinx/ext/viewcode.py", line 115, in collect_pages
    lines[min(end - 1, maxindex)] += '</div>'
IndexError: list index out of range

}}}

I've tried building on two different servers to no avail.

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2010-06-13 17:44:47+00:00_

Thanks, one should remember that list indices go from 0 to len-1 ;)  Fixed in <<cset e4354a1045c5875d56c15674a813f16667a12ec2>>.
